controllers/src/main/scala/cromwell/pipeline/controller/ProjectFileController.scala

```diff
@@ -63,13 +66,21 @@ class ProjectFileController(wdlService: ProjectFileService)(implicit val executi
         }
       },
       path("files" / "configurations") {
-        post {
-          entity(as[ProjectBuildConfigurationRequest]) { request =>
-            onComplete(wdlService.buildConfiguration(request.projectId, request.projectFile)) {
-              case Success(Left(e))              => complete(StatusCodes.UnprocessableEntity, e.errors)
-              case Success(Right(configuration)) => complete(configuration)
-              case Failure(e)                    => complete(StatusCodes.InternalServerError, e.getMessage)
-            }
+        get {
+          parameters('project_id.as[String], 'project_file_path.as[String], 'version.as[String].optional) {
+            (projectId, projectFilePath, version) =>
+              onComplete(
+                wdlService.buildConfiguration(
+                  ProjectId(projectId),
+                  Paths.get(projectFilePath),
+                  version.map(v => PipelineVersion(v)),
```

Comment:
Let's make this deserialization reusable. We can do something like this
```
parameters('project_id.as[ProjectId], 'project_file_path.as[Path], 'version.as[PipelineVersion].optional)
```
We just need to tell Akka how to deserialize our classes. Take a look [here](https://github.com/EpamLifeSciencesTeam/cmwl_pipeline/blob/4fd247669f7118b967306ca2776340d7b0ed22ab/controllers/src/main/scala/cromwell/pipeline/controller/RunController.scala#L24) and [here](https://github.com/EpamLifeSciencesTeam/cmwl_pipeline/blob/4fd247669f7118b967306ca2776340d7b0ed22ab/controllers/src/main/scala/cromwell/pipeline/controller/utils/FromStringUnmarshallers.scala#L9). In this case everyone will be able to reuse it without boilerplate.

Reply:
Thanks. Fixed.